examples/nidcpower_source_dc_voltage/teststand_fixture.py

```diff
@@ -39,48 +44,47 @@ def create_nidcpower_sessions(sequence_context: Any) -> None:
-        session_management_client = nims.session_management.Client(
-            grpc_channel=grpc_channel_pool.session_management_channel
-        )
-
         teststand_support = TestStandSupport(sequence_context)
         pin_map_id = teststand_support.get_active_pin_map_id()
-        pin_map_context = nims.session_management.PinMapContext(pin_map_id=pin_map_id, sites=None)
-        grpc_device_channel = grpc_channel_pool.get_grpc_device_channel(
-            nidcpower.GRPC_SERVICE_INTERFACE_NAME
+        pin_map_context = PinMapContext(pin_map_id=pin_map_id, sites=None)
+
+        discovery_client = DiscoveryClient(grpc_channel_pool=grpc_channel_pool)
+        session_management_client = SessionManagementClient(
+            discovery_client=discovery_client, grpc_channel_pool=grpc_channel_pool
         )
         with session_management_client.reserve_sessions(
-            context=pin_map_context,
-            instrument_type_id=nims.session_management.INSTRUMENT_TYPE_NI_DCPOWER,
+            pin_map_context, instrument_type_id=INSTRUMENT_TYPE_NI_DCPOWER
         ) as reservation:
-            for session_info in reservation.session_info:
-                # Leave session open
-                _ = create_session(
-                    session_info,
-                    grpc_device_channel,
-                    initialization_behavior=nidcpower.SessionInitializationBehavior.INITIALIZE_SERVER_SESSION,
-                )
+            # Initialize new sessions on the server, but do not close them.
+            # (Deliberately leak them.)
+            context_manager = reservation.create_nidcpower_sessions(
+                initialization_behavior=SessionInitializationBehavior.INITIALIZE_SERVER_SESSION
+            )
+            _ = context_manager.__enter__()
+
             session_management_client.register_sessions(reservation.session_info)
 
 
 def destroy_nidcpower_sessions() -> None:
     """Destroy and unregister all NI-DCPower sessions."""
     with GrpcChannelPoolHelper() as grpc_channel_pool:
-        session_management_client = nims.session_management.Client(
-            grpc_channel=grpc_channel_pool.session_management_channel
-        )
-        grpc_device_channel = grpc_channel_pool.get_grpc_device_channel(
-            nidcpower.GRPC_SERVICE_INTERFACE_NAME
+        discovery_client = DiscoveryClient(grpc_channel_pool=grpc_channel_pool)
+        session_management_client = SessionManagementClient(
+            discovery_client=discovery_client, grpc_channel_pool=grpc_channel_pool
         )
         with session_management_client.reserve_all_registered_sessions(
-            instrument_type_id=nims.session_management.INSTRUMENT_TYPE_NI_DCPOWER,
+            instrument_type_id=INSTRUMENT_TYPE_NI_DCPOWER
         ) as reservation:
+            # If there are no registered sessions, there is nothing to clean up.
+            if not reservation.session_info:
+                return
```

Comment:
This is necessary because create_nidcpower_sessions() raises an exception if there are no sessions to create. For measurements, we assume users always want to create one or more sessions, but for code module cleanup it's possible to have zero sessions.

We could potentially tweak the error handling or add an optional parameter to allow creating an empty list.

Reply:
This also avoids making a bunch of grpc calls that are ultimately noops (i.e. unregister_sessions) so this is probably better example code regardless.